packages/nakama-js/socket.ts

```diff
@@ -977,6 +977,12 @@ export class DefaultSocket implements Socket {
     }
   }
 
+  onheartbeattimeout() {
```

Comment:
you'll want to add this to the `Socket` interface as well.

could you also add the following comment to it in the interface ``:

```
An application-level heartbeat timeout that fires after the client does not receive a pong from the server after the heartbeat interval.

Most browsers maintain an internal heartbeat, in which case its unlikely you'll need to use this callback. However, Chrome does not implement an internal heartbeat. 

We fire this separately from `onclose` because heartbeats fail when there's no connectivity, and many browsers don't fire `onclose` until the closing handshake fails. 

In any case, be aware that `onclose` will still fire if there is a heartbeat timeout in a potentially delayed manner.
```

Not the most concise but it's a slightly complicated topic :)